internal/spokes/spokes.go

```diff
@@ -547,7 +547,11 @@ func (r *spokesReceivePack) readPack(ctx context.Context, commands []command, ca
 	// FIXME? add --pack_header similar to git's push_header_arg
 
 	if useSideBand(capabilities) {
-		args = append(args, "--show-resolving-progress", "--report-end-of-input")
+		args = append(args, "--report-end-of-input")
+	}
+
+	if useSideBand(capabilities) && !isQuiet(capabilities) {
```

Comment:
This seems right to me, but it doesn't match [what git does](https://github.com/git/git/blob/950264636c68591989456e3ba0a5442f93152c1a/builtin/receive-pack.c#L2262-L2263). Git only checks `quiet` and ignores whether sideband is enabled for this.

```c
		if (!quiet && err_fd)
			strvec_push(&child.args, "--show-resolving-progress");
```

@derrickstolee @ttaylorr What is the expected behavior if sideband is off and quiet isn't specified?

Reply:
If there's no sideband, then there's nothing we can send the progress to. The `err_fd` being zero actually means "send `stderr` to `stdout`", I think, so we are still doing the right thing here.

Reply:
If `err_fd` is zero, it means that we don't pass `--show-resolving-progress` at all. Like @derrickstolee said, we will send any output that would have gone to stderr to stdout since Git sets the `index-pack` child process to point stderr to `err_fd` regardless of its value.

But we only pass `--show-resolving-progress` when we aren't quiet and have a sideband. So I agree that what is written above is correct.

Reply:
We can't send progress indicators over `stdout` since that is being taken as communication over the Git protocol messages. The same situation for "if there is no sideband, then we can't communicate progress over the main band."